R-TypeGame/SystemNetworkUpdate/UpdatePlayerConnectionSystem/updatePlayerConnectionSystem.cpp

```diff
@@ -8,34 +8,36 @@
 #include "Draw/DrawComponent.hpp"
 #include "updatePlayerConnectionSystem.hpp"
 
-void UpdatePlayerConnectionSystem::_updatePlayerConnection(Network::UDPPacket packet, ECS::Registry& reg)
+void UpdatePlayerConnectionSystem::_updatePlayerConnection(ABINetwork::UDPPacket packet, ECS::Registry& reg)
 {
     std::lock_guard<std::mutex> lock(reg._myBeautifulMutex);
     try {
         uint32_t componentTypeLength = static_cast<size_t>(packet.getPayload()[0]);
 
-        if (packet.getPayload().size() < 1 + componentTypeLength + 4 + sizeof(bool)) {
-            std::cerr << "Payload is too small." << std::endl;
-            return;
-        }
-
         std::string componentType(packet.getPayload().begin() + 1, packet.getPayload().begin() + 1 + componentTypeLength);
 
-        int idxPacketEntities = (packet.getPayload()[1 + componentTypeLength] << 24) |
-                            (packet.getPayload()[2 + componentTypeLength] << 16) |
-                            (packet.getPayload()[3 + componentTypeLength] << 8)  |
-                            packet.getPayload()[4 + componentTypeLength];
+        std::pair<std::string, std::vector<std::variant<int, float, std::string, bool>>> info = ABINetwork::getUpdateComponentInfoFromPacket(packet);
+
+        if (info.second.empty())
+            std::cout << "Vector vide" << std::endl;
 
-        bool drawable;
-        std::memcpy(&drawable, &packet.getPayload()[5 + componentTypeLength], sizeof(bool));
+        int idxEntities = 0;
+        bool drawable = false;
+
+        for (const auto& elem : info.second) {
+            if (std::holds_alternative<int>(elem))

```

Comment:
What does it do ?

Reply:
That check if the type in vector of variant is int